### PR TITLE
chore(packages/prisma): add migration script to compute instance statistics

### PR DESCRIPTION
**Pull request by sjschlapbach**
TESTING - to find suitable instances with few responses for manual validation:

```sql
SELECT ei.id, COUNT(qr.id) FROM "ElementInstance" ei JOIN "QuestionResponse" qr ON ei.id = qr."elementInstanceId"
    WHERE "elementType" = 'SC'
    GROUP BY(ei.id)
```

### CODE REVIEW

File: packages/prisma/src/scripts/2024-08-23_compute_instance_statistics.ts [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+            // set average time spent on instance and total unique participants
+            uniqueParticipantCount: totalUniqueParticipants,
+            averageTimeSpent: averageInstanceTime,
+          },
+        },
+      },
+    })
+  })
+
+  await prisma.$transaction(updates)
```
Comment (rschlaefli) [inline packages/prisma/src/scripts/2024-08-23_compute_instance_statistics.ts]: are all updates to all instances (thousands?) in one transaction like this? I would rather not do such a large transaction and process them separately, as they are independent any way and the computation could just run again and would not change anything

**Review by rschlaefli — APPROVED** [review]

File: packages/prisma/src/scripts/2024-08-23_compute_instance_statistics.ts (hunk 1 above)
Comment (sjschlapbach) [reply packages/prisma/src/scripts/2024-08-23_compute_instance_statistics.ts]: yes, this was not ideal - leftover from copied script :) fixed it

File: packages/prisma/src/scripts/2024-08-23_compute_instance_statistics.ts [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+      case ElementType.FREE_TEXT:
+        const { tempCorrect2, tempIncorrect2 } =
+          instance.anonymousResults.responses.reduce(
+            (acc, response) => {
+              acc.tempCorrect += Number(response.correct)
+              acc.tempIncorrect += Number(!response.correct)
+              return acc
+            },
+            { tempCorrect: 0, tempIncorrect: 0 }
+          )
```
[the remark this answers is not in the document]
Comment (sjschlapbach) [reply packages/prisma/src/scripts/2024-08-23_compute_instance_statistics.ts]: has been replaced through modified implementation in the meantime

File: packages/prisma/src/scripts/2024-08-23_compute_instance_statistics.ts [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+            if (ix === correctAnswerIx) {
+              acc.tempCorrect1 += count as number
+            } else {
+              acc.tempIncorrect1 += count as number
+            }
+
+            return acc
+          },
+          { tempCorrect1: 0, tempIncorrect1: 0 }
+        )
```
[the remark this answers is not in the document]
Comment (sjschlapbach) [reply packages/prisma/src/scripts/2024-08-23_compute_instance_statistics.ts:174]: variable names should be chosen to not interfer with the rest of the code

File: packages/prisma/src/scripts/2024-08-23_compute_instance_statistics.ts [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+            anonymousWrongCount,
+
+            // set average time spent on instance and total unique participants
+            uniqueParticipantCount: totalUniqueParticipants,
+            averageTimeSpent: averageInstanceTime,
+          },
+        },
+      },
+    })
+  })
```
[the remark this answers is not in the document]
Comment (sjschlapbach) [reply packages/prisma/src/scripts/2024-08-23_compute_instance_statistics.ts]: fixed

**Pull request merged**